### PR TITLE
Change the way of getting entries

### PR DESCRIPTION
**Pull request by ninoseki**
I changed to count a number of entries returned from a CT log server and make retries if there is not enough number of entries. (Ref. #22) 

The following are execution results of the original version and my forked version.

### Original version

| time  | number of messages(entries) sent |
|-------|----------------------------------|
| 15:35 | 1627                             |
| 15:36 | 2604                             |
| 15:37 | 2618                             |
| 15:38 | 3309                             |
| 15:39 | 2638                             |
| 15:40 | 2663                             |
| 15:41 | 3099                             |
| 15:42 | 2287                             |
| 15:43 | 3416                             |
| 15:44 | 3136                             |
|       | 27397                            |

- [Raw log file](https://gist.githubusercontent.com/ninoseki/5d79f874bb68c96e7bddabb4e75b742c/raw/7ee89d8ed16c81340a9699716657d0664b410424/original.log)

### Forked version

| time  | number of messages(entries) sent |
|-------|----------------------------------|
| 15:35 | 4115                             |
| 15:36 | 4065                             |
| 15:37 | 4019                             |
| 15:38 | 3943                             |
| 15:39 | 4016                             |
| 15:40 | 4244                             |
| 15:41 | 4941                             |
| 15:42 | 4410                             |
| 15:43 | 6128                             |
| 15:44 | 4291                             |
|       | 44172                            |

- [Raw log file](https://gist.githubusercontent.com/ninoseki/5d79f874bb68c96e7bddabb4e75b742c/raw/7ee89d8ed16c81340a9699716657d0664b410424/forked.log)

In summary, my forked version collected 1.6 times more entries than the original version.

I'm sure that my implementation is not the best. But I think it is better than the current implementation.

### CODE REVIEW

**Comment by Fitblip** [comment]
Hi @ninoseki - Thanks a ton for doing the research here to figure out an issue in the certstream server. I've heard our outputs have dipped in recent months, but I figured that was due to congestion but it appears as though some CT servers have changed how they operate!

I'll pull this down and review it tonight since I have some time, and worst case will spend this weekend getting this solidified and merged.

**Comment by Fitblip** [comment]
Hi @ninoseki - So I did some digging into this. The Google CT servers are responding in very weird ways and I'm not sure why. 

Take this URL for example - https://ct.googleapis.com/logs/argon2020/ct/v1/get-entries?start=341366876&end=341366907, it returns 4 entries, even though we request 31. Yet, when you decrement the `end` parameter, you get 31 entries https://ct.googleapis.com/logs/argon2020/ct/v1/get-entries?start=341366876&end=341366906. No idea what's going on, but I've reached out to someone on the CT team at Google (though it's been a few years since we've spoken) to potentially flag the issue to them. 

Either way, while testing this I ended up building out an alternative implementation that I opened as a PR - https://github.com/CaliDog/certstream-server/pull/24. If you have a moment, do you mind running your testing script against that and/or reviewing it for any bugs/issues you see? 

Thanks again for putting this PR in, much appreciated 🎉

**Comment by Fitblip** [comment]
Ah this is actually just a bug caused by my break in expectations from the RFC. I originally figured that 64 entries was a reasonable number which seemed to work fine in my testing, but the RFC specifies that the server MAY honor the ranges, and it might just do whatever it wants when given a range - https://tools.ietf.org/html/rfc6962#section-4.6.

**Comment by ninoseki** [comment]
Wow, #24 looks great! 
Absolutely the batch-size based processing is better than my PR.
